### ISSUE TITLE
Need to update conda-forge recipe

### ISSUE THREAD

**Issue by zhang-alvin** [erdc/proteus#1082]
conda-forge version is 1.7.1 which encounters the following error in `proteus_notebooks/Tutorials/Tutorial_1/Introduction.ipynb`

```
# scotch has been compiled with SCOTCH_PTHREAD
# so MPI_Init() must be called with MPI_THREAD_MULTIPLE, or MPI_Init_thread() must be called.
# at import time, mpi4py initializes the MPI execution environment calling MPI_Init_thread().
# mpi4py must be imported before petsc4py otherwise petsc4py initializes MPI with MPI_Init().
```

**Comment by zhang-alvin** [comment]
need to remove .h5 references in setup.py file

**Comment by davidbrochart** [comment]
> conda-forge version is 1.7.1 which encounters the following error in `proteus_notebooks/Tutorials/Tutorial_1/Introduction.ipynb`
> 
> ```
> # scotch has been compiled with SCOTCH_PTHREAD
> # so MPI_Init() must be called with MPI_THREAD_MULTIPLE, or MPI_Init_thread() must be called.
> # at import time, mpi4py initializes the MPI execution environment calling MPI_Init_thread().
> # mpi4py must be imported before petsc4py otherwise petsc4py initializes MPI with MPI_Init().
> ```

This can be solved by importing mpi4py first thing, see #1047.

**Comment by zhang-alvin** [comment]
@davidbrochart Yes this was resolved already with the previous release. If I recall correctly, you had run into the problem earlier and resolved it on the master branch, but it was not included in one of the releases. conda-forge/proteus-feedstock#18 resolved this issue.
